token/cli/src/main.rs

```diff
@@ -3955,12 +3991,39 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    #[serial]
+    async fn balance_with_owner() {
+        let (test_validator, payer) = new_validator_for_test().await;
+        for program_id in [spl_token::id(), spl_token_2022::id()] {
+            let config = test_config_with_default_signer(&test_validator, &payer, &program_id);
+            let token = create_token(&config, &payer).await;
+            let _account = create_associated_account(&config, &payer, token).await;
+            let config = test_config_without_default_signer(&test_validator, &program_id);
+            let result = process_test_command(
+                &config,
+                &payer,
+                &[
+                    "spl-token",
+                    CommandName::Balance.into(),
+                    &token.to_string(),
+                    "--owner",
+                    &payer.pubkey().to_string(),
+                ],
+            )
+            .await;
+            let value: serde_json::Value = serde_json::from_str(&result.unwrap()).unwrap();
+            assert_eq!(value["amount"], "0");
+            assert_eq!(value["uiAmountString"], "0");
+        }
+    }
```

Comment:
This test seems to be just about the same thing as `balance_after_mint_with_owner`, just without the minting. Since you're testing non-zero values in the other test, I'd recommend removing this one